arangod/GeneralServer/GeneralCommTask.cpp

```diff
@@ -75,17 +75,24 @@ void GeneralCommTask<T>::asyncReadSome() {
   // first try a sync read for performance
   if (_protocol->supportsMixedIO()) {
     std::size_t available = _protocol->available(ec);
-    while (!ec && available > 8) {
+    bool cont = true;
+    while (!ec && available > 8 && cont) {
       auto mutableBuff = _protocol->buffer.prepare(available);
       size_t nread = _protocol->socket.read_some(mutableBuff, ec);
       _protocol->buffer.commit(nread);
       if (ec) {
         break;
       }
-      if (!readCallback(ec)) {
-        return;
+      
+      try {
+        cont = readCallback(ec);
+        available = _protocol->available(ec);
+      } catch (...) {
+        LOG_TOPIC("2c6b5", ERR, arangodb::Logger::REQUESTS)
```

Comment:
I feel that this could also justify `FATAL` - since callbacks shouldn't let exceptions out?

Reply:
but it is not fatal

Reply:
if maintainer mode?